File: emulinker/src/main/java/org/emulinker/kaillera/model/event/PlayerDesynchEvent.java
package org.emulinker.kaillera.model.event;

import org.emulinker.kaillera.model.*;

public class PlayerDesynchEvent implements GameEvent {
  private KailleraGame game;
  private KailleraUser user;
  private String message;

  public PlayerDesynchEvent(KailleraGame game, KailleraUser user, String message) {
    this.game = game;
    this.user = user;
    this.message = message;
  }

  @Override
  public String toString() {
    return "GameDesynchEvent";
  }

  @Override
  public KailleraGame getGame() {
    return game;
  }

  public KailleraUser getUser() {
    return user;
  }

  public String getMessage() {
    return message;
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/event/PlayerDesynchEvent.kt
package org.emulinker.kaillera.model.event

import org.emulinker.kaillera.model.KailleraGame
import org.emulinker.kaillera.model.KailleraUser

class PlayerDesynchEvent(
    override val game: KailleraGame, val user: KailleraUser, val message: String
) : GameEvent {
  override fun toString(): String {
    return "GameDesynchEvent"
  }
}
